src/test/java/com/devonfw/sample/archunit/SecurityTest.java

```diff
@@ -0,0 +1,34 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+
+import javax.annotation.security.DenyAll;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.methods;
+
+/**
+ * JUnit test that validates the security rules of this application.
+ */
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class SecurityTest {
+
+    /**
+     * Checks 'UcImpl' classes for public methods.
+     * Fails if a method is neither annotated with @PermitAll, @RolesAllowed nor @DenyAll.
+     */
+    @ArchTest
+    private static final ArchRule shouldBeProperlyAnnotated = //
+            methods()
+                    .that().areDeclaredInClassesThat().haveSimpleNameEndingWith("UcImpl")
```

Comment:
use-cases start with `Uc`. In the classic structure we separated interface ( e.g. `UcFindTask`) from implementation (e.g. `UcFindTaskImpl`) but this separation was dropped as in reality there is only one implementation for that interface and therefore the separation is only waste.
Here is an example in our app:
https://github.com/devonfw-sample/archunit/blob/master/src/main/java/com/devonfw/sample/archunit/task/logic/UcDeleteTaskItem.java#L21
So we could either check that the name starts with `Uc` or the class extends `AbstractUc`.

Please note that this will break as the security annotations are all commented out.
Could you even uncomment these annotations within your PR? This will currently prevent us from testing the app as we cannot start and access it anymore without any user-management (that is why I so far commented them out) but for a realistic app you would have these annotations active and that is what we want to demonstrate...

Reply:
Thanks! I have fixed the class namecheck and uncommented all security annotations. Please notify me if I have missed anything else.